.github/workflows/slow.yml

```diff
@@ -19,30 +19,56 @@ env:
 
 jobs:
   Tests:
+    name: "Tests (os: ${{ matrix.os }}, flags: \"${{ matrix.flags }}\") (partition: ${{ matrix.partition }}/4)"
     runs-on: ${{ matrix.os }}
     timeout-minutes: 180
     strategy:
       matrix:
-        include:
-          - os: ubuntu-latest
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        partition: [1, 2, 3, 4]
+        flags:
+          - "--features commonware-runtime/iouring-storage"
+          - "--features commonware-runtime/iouring-network"
+          - "--no-default-features"
+          - ""
+        exclude:
+          # Only run io_uring features on Ubuntu
+          - os: windows-latest
+            flags: "--features commonware-runtime/iouring-storage"
+          - os: macos-latest
             flags: "--features commonware-runtime/iouring-storage"
-          - os: ubuntu-latest
+          - os: windows-latest
             flags: "--features commonware-runtime/iouring-network"
-          - os: ubuntu-latest
-            flags: "--no-default-features"
-          - os: ubuntu-latest
-            flags: ""
+          - os: macos-latest
+            flags: "--features commonware-runtime/iouring-network"
+          # Only run --no-default-features on Ubuntu
           - os: windows-latest
-            flags: ""
+            flags: "--no-default-features"
           - os: macos-latest
-            flags: ""
+            flags: "--no-default-features"
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
     - name: Run setup
       uses: ./.github/actions/setup
+    - name: Install nextest
+      uses: taiki-e/install-action@cargo-nextest
     - name: Run ignored tests
-      run: cargo test ${{ matrix.flags }} --verbose -- --ignored
+      run: cargo nextest run ${{ matrix.flags }} --partition hash:${{matrix.partition}}/4 --verbose -- --ignored
+
+  Tests-Gate:
+    name: "Slow-Tests"
+    runs-on: ubuntu-latest
+    needs: Tests
+    if: always()
+    steps:
+      - name: Check test results
+        run: |
+          if [ "${{ needs.Tests.result }}" != "success" ]; then
```

Comment:
If we do exit on failure (I think that's a mode), I was under the impression `needs` would fail?

Reply:
It shouldn't per [the docs](https://docs.github.com/en/actions/concepts/workflows-and-actions/workflows#needs-context), the object should be able to give us information about a failure alongside `if: always()`.